aws/rds_snapshot_test.go

```diff
@@ -0,0 +1,134 @@
+package aws
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/rds"
+
+	"github.com/gruntwork-io/cloud-nuke/config"
+	"github.com/gruntwork-io/cloud-nuke/util"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// createTestDBInstance generates a test DBInstance since snapshots can only be created
+// from an existing DB Instance
+func createTestDBInstance(t *testing.T, session *session.Session, name string) {
+	svc := rds.New(session)
+	input := &rds.CreateDBInstanceInput{
+		AllocatedStorage:     awsgo.Int64(5),
+		DBInstanceClass:      awsgo.String("db.m5.large"),
+		DBInstanceIdentifier: awsgo.String(name),
+		Engine:               awsgo.String("postgres"),
+		MasterUsername:       awsgo.String("gruntwork"),
+		MasterUserPassword:   awsgo.String("password"),
+	}
+
+	_, err := svc.CreateDBInstance(input)
+	require.NoError(t, err)
+
+	svc.WaitUntilDBInstanceAvailable(&rds.DescribeDBInstancesInput{
+		DBInstanceIdentifier: &name,
+	})
+}
+
+//createTestRDSSnapshot generates a test DB Instance Snapshot
+func createTestRDSSnapshot(t *testing.T, session *session.Session, instanceName string, snapshotName string) {
+	svc := rds.New(session)
+	input := &rds.CreateDBSnapshotInput{
+		DBInstanceIdentifier: awsgo.String(instanceName),
+		DBSnapshotIdentifier: awsgo.String(snapshotName),
+	}
+
+	_, err := svc.CreateDBSnapshot(input)
+	require.NoError(t, errors.WithStackTrace(err))
+
+	svc.WaitUntilDBSnapshotAvailable(&rds.DescribeDBSnapshotsInput{
+		DBInstanceIdentifier: &instanceName,
+		DBSnapshotIdentifier: &snapshotName,
+	})
+
+	result, err := svc.DescribeDBSnapshots(&rds.DescribeDBSnapshotsInput{
```

Comment:
Check `err`! 

BTW, many editors with good Golang support can warn you about ignored errors automatically.

Reply:
Fixed